modules/jupyter-widget/src/widget.js

```diff
@@ -125,9 +125,23 @@ export class DeckGLView extends DOMWidgetView {
 
     this.model.on('change:json_input', this.valueChanged.bind(this), this);
     this.model.on('change:data_buffer', this.dataBufferChanged.bind(this), this);
+    this.model.on('change:classes', this.addNewCustomLayers.bind(this), this);
+
+    this.addNewCustomLayers();
     this.dataBufferChanged();
   }
 
+  addNewCustomLayers() {
+    if (this.model.get('classes')) {
+      const customLibraries = this.model.get('classes');
+      for (const obj of customLibraries) {
+        // obj is an object of {libraryName: <string>, url: <string>}
+        const [libraryName, resourceUri] = Object.entries(obj)[0];
```

Comment:
`obj.libraryName`,
`obj.url`

Reply:
They're arriving as a list of dictionaries from the server in something like this format:

`[{"libraryName1": "url1"}, {"libraryName2": "url2"}]`

Reply:
Don’t you have control of the format on the server side?

Reply:
I do but I'm personally partial to the more concise form of 

`[{"libraryName1": "url1"}, {"libraryName2": "url2"}]`

instead of

`[{"libraryName": "libraryName1", "resourceUri": "url1"}, {"libraryName": "libraryName2", "resourceUri": "url2"}]`

It's anyone's choice, however–happy to switch it over.

Reply:
Think of each item as an entry in a database. Predictable data structured makes your code a lot more readable. Using the key name as the library name is implicit, and require additional documentation in the source. Explicitly calling out `libraryName` leaves no ambiguity.

If you insist on using keys as library names, which is fine, then there's no reason to have an array. All entries can be merged into one dictionary. `Object.entries(obj)[0]` is a bad pattern because you are relying on the existence and uniqueness of the object's content. It will easily crash if the input is malformed.